src/main/java/appeng/helpers/IInterfaceTerminalSupport.java

```diff
@@ -0,0 +1,38 @@
+package appeng.helpers;
```

Comment:
I am not too confident about this. Despite miozune's suggestion, I think it is more appropriate to reuse an existing interface, such as `ICraftingProvider` which has the properties we want.

Reply:
As long as all new methods have a `default` implementation, it's backwards compatible

Reply:
| Part Name  | ICraftingProvider | Support Interface Terminal | IInterfaceHost |
| - | - | - | - |
| Level Emitter | ○ | × | × |
| Crafting Input Buffer  | ○ | ○ | × |
| ME Interface | ○ | ○ | ○ |

`ICraftingProvider` does not necessarily have ability to support interface terminal. Having separated interface contributes to reducing unnecessary methods declared on `ICraftingProvider`. If needed we can make `IInterfaceTerminalSupport` extend `ICraftingProvider`.